src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -2030,19 +2030,19 @@ internal static string ConcatenateStringPathArguments(CommandElementAst stringAs
             {
                 try
                 {
-                    if (argumentCompleterAttribute.Type != null)
+                    var completer = argumentCompleterAttribute.Type != null
+                        ? Activator.CreateInstance(argumentCompleterAttribute.Type) as IArgumentCompleter
+                        : argumentCompleterAttribute is IArgumentCompleterFactory factory ? factory.Create() : null;
```

Comment:
Interesting, can we put all in a Factory?
I mean something like:
```c#
var completer = Factory.Create(argumentCompleterAttribute)
```

Reply:
Adding method to ArgumentCompleterAttribute